_posts/2020-02-24-qute.adoc

```diff
@@ -0,0 +1,182 @@
+====
+
+Let's start with a very good question: _"Why yet another templating engine?"_.
+There are plenty of templating libraries in Java.
+And Quarkus is known to build on top of "Best of Breed Libraries and Standards".
+That's true.
+On the other hand, Quarkus community is also a powerful innovation catalyst.
+And so we decided to start Qute (QUarkus TEmplates) - a templating engine designed specifically to meet the Quarkus needs. 
+We believe that we can bring new ideas even to such an explored area as the templating is.
+
+=== Basic Ideas
+
+Our main goal is to provide an opinionated innovative templating engine.
+But we don't want to reinvent the wheel.
+Instead, we got inspired by existing technologies.
+Just a few examples:
+
+* The https://quarkus.io/guides/qute-reference#syntax-and-building-blocks[syntax] is mainly inspired by https://handlebarsjs.com/[Handlebars] and https://www.dustjs.com/[Dust.js].
+* The https://quarkus.io/guides/qute-reference#include_helper[template inheritance] is inspired by https://en.wikipedia.org/wiki/Facelets[Facelets] and https://docs.djangoproject.com/en/3.0/ref/templates/language/[Django]. 
+* Qute supports the https://en.wikipedia.org/wiki/Elvis_operator[elvis operator] you might be familiar with from https://kotlinlang.org/[Kotlin]. 
+* https://quarkus.io/guides/qute-reference#template_extension_methods[Extension methods]  that can be used to extend the data classes with new functionality are also inspired by modern languages.
+* If you come from the world of JSP/JSF/Facelets you'll appreciate that `@Named` CDI beans can be referenced directly in any template through the `inject` namespace, e.g. `{inject:foo.price}`. See https://quarkus.io/guides/qute-reference#injecting-beans-directly-in-templates[Injecting Beans Directly In Templates] for more information.
+
+But that's not all.
+We do introduce new features based on Quarkus principles.  
+
+=== Asynchronous Data Resolution - On The Way To Reactive
+
+When we started to design Qute we had one important aspect in mind - the data resolution API should be asynchronous.
+This allows for better resource utilization and fits the Quarkus reactive story.
+Another useful consequence is that it's possible to use a non-blocking client directly from a template to fetch data from external systems:
+
+.Non-blocking Client Data Fetching Example
+[source,html]
+----
+{@org.acme.Client client} <1>
+<html>
+<body>
+    <h1>Quarkus Open Pull Requests</h1>
+    {#for pull in client.pullRequests} <2>
+        <p>{pull.title} - {pull.user.login}</p>
+    {/for} 
+</body>
+</html>
+----
+<1> Parameter declaration - maps `client` to `org.acme.Client`. See the <<type-safe-templates,next section>> for more information.
+<2> `org.acme.Client#getPullRequests()` is using a non-blocking Vert.x client to fetch the data directly from the GitHub API. Since the data resolution is asynchronous the thread is not blocked and can continue performing some other tasks.
```

Comment:
It would be better to just show the API of getPullRequests and show the non blocking type

Reply:
I will add a snippet with the `getPullRequests()` pseudo-body.